spec/syntax.md

```diff
@@ -905,24 +926,64 @@ backslash       = %x5C ; U+005C REVERSE SOLIDUS "\"
 
 ### Whitespace
 
-**_<dfn>Whitespace</dfn>_** is defined as one or more of
-U+0009 CHARACTER TABULATION (tab), 
-U+000A LINE FEED (new line),
-U+000D CARRIAGE RETURN, 
-U+3000 IDEOGRAPHIC SPACE, 
-or U+0020 SPACE.
+The syntax limits whitespace characters outside of a _pattern_ to the following:
+`U+0009 CHARACTER TABULATION` (tab), 
+`U+000A LINE FEED` (new line),
+`U+000D CARRIAGE RETURN`, 
+`U+3000 IDEOGRAPHIC SPACE`, 
+or `U+0020 SPACE`.
 
 Inside _patterns_ and _quoted literals_,
 whitespace is part of the content and is recorded and stored verbatim.
 Whitespace is not significant outside translatable text, except where required by the syntax.
 
+There are two whitespace productions in the syntax.
+**_<dfn>Optional whitespace</dfn>_** is whitespace that is not required by the syntax, 
+but which users might want to include to increase the readability of a _message_.
+**_<dfn>Required whitespace</dfn>_** is whitespace that is required by the syntax.
+
+Tools or users SHOULD insert paired isolating controls `U+2066 LEFT-TO-RIGHT ISOLATE`
```

Comment:
- The SHOULD should be limited to those pieces that contain RTL text.
- The SHOULD should be limited to "tooling".
I think "or users" sets unrealistic expectations; manual insertion can be tricky, and runs the risk with isolates of having unpaired bidi controls.
Maybe instead add at the end a note that users may also manually insert these characters, 
but tooling can make sure that the bidi controls are in the right positions,
and that all and only the bidi controls necessary are included.
- The two sections about inserting isolates and inserting marked should be clearly separated.

Reply:
@macchiati 

The SHOULD is already limited to RTL. I added an RTL part of the note about literals.

I split the isolate and marks paragraphs apart.

There are two ways to go on the "tools or users" thing. An alternative to getting into who should do it is to say "Messages SHOULD include..." and "Messages MAY include..." and then having a note about tools, which is how I've handled this.

Note that the syntax requires expression and quoted-pattern controls to be paired (although the whitespace rules break this)